litmus-portal/frontend/src/views/CreateWorkflow/CustomWorkflow/CreateWorkflow/index.tsx

```diff
@@ -0,0 +1,328 @@
+    {
+      onCompleted: (data) => {
+        data.getCharts.forEach((data) => {
+          return data.Spec.Experiments?.forEach((experiment) => {
+            allExp.push({
+              ChaosName: data.Metadata.Name,
+              ExperimentName: experiment,
+            });
+          });
+        });
+        setAllExperiment(allExp);
+      },
+      fetchPolicy: 'cache-and-network',
+    }
+  );
+  // Function to get charts of a particular hub
+  const findChart = (hubname: string) => {
+    const myHubData = data?.getHubStatus.filter((myHub) => {
+      return hubname === myHub.HubName;
+    })[0];
+    getCharts({
+      variables: {
+        data: {
+          UserName: userData.username,
+          RepoURL: myHubData?.RepoURL,
+          RepoBranch: myHubData?.RepoBranch,
+          HubName: hubname,
+        },
+      },
+    });
+    setSelectedHubDetails(myHubData);
+    workflowAction.setWorkflowDetails({
+      customWorkflow: {
+        ...workflowDetails.customWorkflow,
+        hubName: hubname,
+        repoUrl: myHubData?.RepoURL,
+        repoBranch: myHubData?.RepoBranch,
+      },
+    });
+  };
+
+  useEffect(() => {
+    if (selectedHub === 'Public Hub') {
+      setSelectedHub('Public Hub');
+      const ChartsData = hubData.charts;
+      ChartsData.forEach((data: Chart) => {
+        if (data.Spec.Experiments) {
+          data.Spec.Experiments.forEach((experiment) => {
+            allExp.push({
+              ChaosName: data.Metadata.Name,
+              ExperimentName: experiment,
+            });
+          });
+        }
+      });
+      setAllExperiment([...allExp]);
+      workflowAction.setWorkflowDetails({
+        customWorkflow: {
+          ...workflowDetails.customWorkflow,
+          hubName: 'Public Hub',
+          repoUrl: 'https://github.com/litmuschaos/chaos-charts',
+          repoBranch: 'master',
+        },
+      });
+    } else {
+      setAllExperiment([]);
+    }
+  }, [selectedHub]);
+  const availableHubs: MyHubDetail[] = data ? data.getHubStatus : [];
+
+  return (
+    <div className={classes.root}>
+      <div className={classes.headerDiv}>
+        <BackButton isDisabled={false} />
+        <Typography variant="h3" className={classes.headerText} gutterBottom>
+          {t('customWorkflow.createWorkflow.create')}
+        </Typography>
+        <Typography className={classes.headerDesc}>
+          {t('customWorkflow.createWorkflow.createDesc')}
+        </Typography>
+      </div>
+      <div className={classes.workflowDiv}>
+        <Typography variant="h4">
+          <strong> {t('customWorkflow.createWorkflow.workflowInfo')}</strong>
+        </Typography>
+        <div>
+          <div className={classes.inputDiv}>
+            <Typography variant="h6" className={classes.titleText}>
+              {t('customWorkflow.createWorkflow.workflowName')}:
+            </Typography>
+            <InputField
+              label="Workflow Name"
+              styles={{
```

Comment:
Consider adding to the styles.ts file, or using the updated kubera-ui: "^0.2.0" version which includes the InputField component.

Reply:
@vanshBhatia-A4k9 This can't be added to styles.ts since it is a prop that the input filed component takes. Yes, we will be using the kubera-ui once all the components are there, we will be refactoring the litmus-portal with kubera-ui later, since it would be a breaking change.